bun/whatsapp-with-vonage/src/main.ts

```diff
@@ -0,0 +1,74 @@
+import jwt from "jsonwebtoken";
+import { fetch, Headers } from "fetch-undici";
+
+export default async ({ req, res, log, error }) => {
+  if (req.method === "GET") {
+    const fileName = Bun.resolveSync("../static/index.html", import.meta.dir);
+    const html = await Bun.file(fileName).text();
+
+    return res.send(html, 200, {
+      "Content-Type": "text/html; charset=utf-8",
+    });
+  }
+
+  if (req.method === "POST") {
+    const secret = `${Bun.env.VONAGE_API_KEY}:${Bun.env.VONAGE_ACCOUNT_SECRET}`;
+    const basicAuthToken = btoa(secret);
+    const authHeader = req.headers.authorization.split(" ")[1];
+    await jwt.verify(
+      authHeader,
+      Bun.env.SIGNATURE_SECRET,
+      { algorithms: ["HS256"] },
+      (error, decodedToken) => {
+        if (error) {
+          return res.json(
+            {
+              ok: false,
+              error: "can't verify",
+            },
+            400
+          );
+        }
+      }
+    );
```

Comment:
This is quite complicated. Take a look the [node template](https://github.com/appwrite/templates/blob/6dab35f1f221ba075640d93f5a8e374f4fba4349/node/whatsapp-with-vonage/src/main.js#L20)
```js
  const token = (req.headers.authorization ?? '').split(' ')[1];
  var decoded = jwt.verify(token, process.env.VONAGE_API_SIGNATURE_SECRET, {
    algorithms: ['HS256'],
  });

  if (sha256(req.bodyRaw) != decoded['payload_hash']) {
    return res.json({ ok: false, error: 'Payload hash mismatch.' }, 401);
  }
```